Network Routers REST API

gildub opened the pull request:
Add capability to retrieve network routers directly

imtayadeway commented:
@gildub this looks promising, but can you add some tests to prove that this works?

gildub commented:
@imtayadeway, I added the test file, thanks.

gildub commented:
@imtayadeway,

* I've removed the subcollection
* I've added tests for the bulk query in `spec/requests/api/collections_spec.rb` for the query itself (200) and in `spec/requests/api/network_routers_spec.rb` for the (403)
* I've also added the default query in `spec/requests/api/collections_spec.rb`  which I believe makes parts of `spec/requests/api/network_routers_spec.rb` redundant

imtayadeway commented on spec/requests/api/collections_spec.rb:
```diff
@@ -134,6 +134,11 @@ def test_collection_bulk_query(collection, collection_url, klass, id = nil)
       test_collection_query(:hosts, hosts_url, Host, :guid)
     end
 
+    it 'queries NetworkRouters' do
```
Similar question to https://github.com/ManageIQ/manageiq/pull/15524/files#r127473371

gildub replied:
@imtayadeway, sure although I believe all those test could go in the collection but that would be the purpose of another PR ;)

abellotti commented:
Thanks @gildub for the API Enhancement.  LGTM !!
